index.js

```diff
@@ -24,7 +24,7 @@ module.exports = function(options) {
       return callback(new Error('Streaming not supported.'));
     }
 
-    if (file.stat.isDirectory()) {
+    if (!file.isBuffer && file.stat.isDirectory()) {
```

Comment:
Actually, it should be [file.isDirectory](https://github.com/gulpjs/vinyl/blob/master/index.js#L46)